src/main/java/com/google/javascript/gents/TypeConversionPass.java

```diff
@@ -418,6 +419,17 @@ void convertDefineClassToClass(Node n) {
     compiler.reportCodeChange();
   }
 
+  boolean inGoogDefineClass(Node n) {
+    if (n == null) {
+      return false;
+    }
+
+    if (n.isCall() && "goog.defineClass".equals(n.getFirstChild().getQualifiedName())) {
+      return true;
+    }
+
+    return inGoogDefineClass(n.getParent());
```

Comment:
I would skip traversing up here. What if you have something deeply nested what kinda made sense (I am making up stuff, but you never know):

```
goog.defineClass(null, {
  foo: function() {
    callMyApiThatExpectsAnonymousClass(/** @constructor */ function () {});
});
```

I think this should be a bit more restrictive and it should also check that the prop name is 'constructor'.